### DISCUSSION TITLE
[Application Development] Interacting with L2 chains

### DISCUSSION THREAD

**Discussion by ohbyeongmin** [ethereum-optimism/developers#339] [Application Development]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Hi! i have a question.
I developing a project in golang using optimism package. I want to use txManager for interact with op-geth. But txManager has only L1RPCURL config. so i thought, are there any other utilities that interact with L2 chains?
What I'm wondering is, does have another utilities better than txManager for interact with op-geth? and If you know of any other utilities, please recommend them

### Additional Information

i use optimism v1.7.2

**Comment by imtipi** [comment]
https://github.com/cartesi/tx-manager
are you talking about this?i didn't see the L1RPCURL things

> What I'm wondering is, does have another utilities better than txManager for interact with op-geth? and If you know of any other utilities, please recommend them

web3js web3py etherjs

**Reply by Chomtana** [reply]
I think he means TxManager in this file

https://github.com/ethereum-optimism/optimism/blob/develop/op-service/txmgr/txmgr.go

**Comment by Chomtana** [comment]
What's your use case for interacting with op-geth? If you are looking for a simple transaction management system for signing and publishing transactions, you should use `github.com/ethereum/go-ethereum` package itself

Example: https://stackoverflow.com/questions/45558034/build-raw-ethereum-transaction-in-go-contract-function-call
